hq/markdown/snyk.md

```diff
@@ -0,0 +1,227 @@
+
+If there is a known non-vulnerable version listed, you can simply change the dependency declaration as below:
+
+```
+$ git diff build.sbt
+diff --git a/build.sbt b/build.sbt
+index 434dbb2..d123098 100644
+--- a/build.sbt
++++ b/build.sbt
+@@ -29,7 +29,11 @@ libraryDependencies ++= Seq(
+   "org.typelevel" %% "cats" % "0.8.1",
+-  "my.library" %% "cleverstuff" % "0.0.99"
++  "my.library" %% "cleverstuff" % "1.0.0+"
+ )
+
+```
+
+## Alternatives
+
+If there is no non-vulnerable version of the library, consider using an alternative.  This, of course, is likely to mean more
+work altering the calling code, and so is not the preferred solution.
+
+```
+$ git diff build.sbt
+diff --git a/build.sbt b/build.sbt
+index 434dbb2..d123098 100644
+--- a/build.sbt
++++ b/build.sbt
+@@ -29,7 +29,11 @@ libraryDependencies ++= Seq(
+   "org.typelevel" %% "cats" % "0.8.1",
+-  "my.library" %% "quickanddirty" % "0.0.3"
++  "his.library" %% "polished" % "1.1.0+"
+ )
+
+```
+
+# Exemptions
+
+The worst case scenario is that the library is still vulnerable, has no alternatives, and cannot be removed.  You
+may wish to build and release anyway, ideally reporting the situation to a risk register. This is most likely to happen when
+a new vulnerability is discovered and the library publisher has not had chance to respond.
+
+This is cleanly achieved by creating a `.snyk` file which can act both as an exemption and the risk register itself.
+Please try to give meaningful reasons for allowing the exemption.
+
+The `.snyk` file can then be added to the repository and the build should continue.
+
+When an exemption expires, the build will start to fail again (see Reviewing Expired Exemptions below).
+If a review still finds no mitigation available, then it is trivial to extend by changing the date and committing.
+Therefore repeated exemptions of no more than a month or two are preferred over a single long exemption.
+
+## `.snyk` file format
+
+See https://support.snyk.io/frequently-asked-questions/finding-vulnerabilities/how-can-i-ignore-a-vulnerability
+
+```
+version: v1.5.0
+ignore:
+  VULNERABILITY-LABEL:
+    - '* > PACKAGE-PROVIDER:PACKAGE':
+      reason: REASON
+      expires: ISO8601-DATE
+```
+eg
+```
+version: v1.5.0
+  'SNYK-JAVA-XERCES-31497':
+    - '* > xerces:xercesImpl':
+      reason: 'Vulnerability requires local access, and this is an internal service'
+      expires: 2018-01-11T00:00:00Z
+  'SNYK-JAVA-ORGAPACHEHTTPCOMPONENTS-31517':
+    - '* > org.apache.httpcomponents:httpclient':
+      reason: 'No remediation available; moving to HTTPS anyway'
+      expires: 2018-01-11T00:00:00Z
+```
+
+Note that the entry includes an expiry date.  Good practice would suggest that this should be of reasonable short duration.
+Long expiry times will mask problems.
+
+## Reviewing Expired Exemptions
+
+When reviewing accepted risks, it would be prudent to examine the git history to discover how long a problem has existed.
```

Comment:
In particular here, while teams are responsible for the security of their own products, InfoSec are the best place to go for guidance on what would be appropriate.